src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/ReceiveCCMNextTime.hpp

```diff
@@ -0,0 +1,49 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <tuple>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"
+
+namespace GeneralizedHarmonic {
+namespace Actions {
+
+template <typename Metavariables>
+struct ReceiveCCMNextTime {
+  using inbox_tags = tmpl::list<Cce::ReceiveTags::CCMNextTimeToGH>;
+  template <typename DbTags, typename... InboxTags, typename ArrayIndex,
+            typename ActionList, typename ParallelComponent>
+  static Parallel::iterable_action_return_t apply(
+      db::DataBox<DbTags>& box, tuples::TaggedTuple<InboxTags...>& inboxes,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    auto& inbox = tuples::get<Cce::ReceiveTags::CCMNextTimeToGH>(inboxes);
+
+    const auto& element = db::get<domain::Tags::Element<3_st>>(box);
+    if (element.external_boundaries().size() == 0_st) {
+      inbox.clear();
+      return {Parallel::AlgorithmExecution::Continue, std::nullopt};
+    }
+    if (inbox.count(db::get<::Tags::TimeStepId>(box)) != 1) {
+      return {Parallel::AlgorithmExecution::Retry, std::nullopt};
+    }
+
+    db::mutate<Cce::Tags::NextCCMTime>(
+        make_not_null(&box),
+        [&inbox](const gsl::not_null<double*> next_time,
+                 const TimeStepId& time) { *next_time = inbox[time]; },
+        db::get<::Tags::TimeStepId>(box));
```

Comment:
And this would be

```cpp
db::mutate<Cce::Tags::NextCCMTime>(
        make_not_null(&box),
        [&next_time](const gsl::not_null<double*> next_ccm_time) {
            *next_ccm_time = next_time;
        });
```